### PR TITLE
Check all snapshots in SnapshotLifecycleRestIT.testFullPolicy

### PR DESCRIPTION
**Pull request by dakrone**
Rather than check the first returned snapshot for a snapshot starting with `snap-` in
SnapshotLifecycleRestIT.testFullPolicy, this commit changes the test to find any snapshots starting
with `snap-`.

In the event that there are no snapshots (the failure case), this also exposes the full results map
so we can diagnose why a failure occurred.

Relates to #50358

### CODE REVIEW

**Review by andreidan — APPROVED** [review]
LGTM, thanks @dakrone

File: x-pack/plugin/ilm/qa/multi-node/src/test/java/org/elasticsearch/xpack/slm/SnapshotLifecycleRestIT.java [hunk 1]
```diff
@@ -108,9 +109,11 @@ public void testFullPolicySnapshot() throws Exception {
             List<Map<String, Object>> snapResponse = ((List<Map<String, Object>>) snapshotResponseMap.get("responses")).stream()
                 .findFirst()
                 .map(m -> (List<Map<String, Object>>) m.get("snapshots"))
+                .filter(allRepos -> allRepos.stream()
+                    .anyMatch(snapMap -> Optional.ofNullable((String) snapMap.get("snapshot"))
+                        .map(snapName -> snapName.startsWith("snap-"))
+                        .orElse(false)))
                 .orElseThrow(() -> new AssertionError("failed to find snapshot response in " + snapshotResponseMap));
```
Comment (probakowski) [inline x-pack/plugin/ilm/qa/multi-node/src/test/java/org/elasticsearch/xpack/slm/SnapshotLifecycleRestIT.java]: It's very hard for me to understand this, mix of streams, `Optional` and nested checks with streams doesn't work well for me. 
I'd rather split it into separate calls which is more explicit and it makes this code easier to digest:
```java
Map<String, Object> snapResponse;
try {
    Map<String, Object> responseMap = ((List<Map<String, Object>>) snapshotResponseMap.get("responses")).get(0);
    List<Map<String, Object>> snapshots = (List<Map<String, Object>>) responseMap.get("snapshots");
    assertTrue(snapshots.stream().anyMatch(s -> s.containsKey("snapshot") && s.get("snapshot").toString().startsWith("snap-")));
    snapResponse = snapshots.get(0);
} catch (Exception e){
    throw new AssertionError("failed to find snapshot response in " + snapshotResponseMap, e);
}
assertThat(snapResponse.get("indices"), equalTo(Collections.singletonList(indexName)));
Map<String, Object> metadata = (Map<String, Object>) snapResponse.get("metadata");
```
If you go with above, please check if I got it right, as I said it's hard for me to follow.

Comment (dakrone) [reply x-pack/plugin/ilm/qa/multi-node/src/test/java/org/elasticsearch/xpack/slm/SnapshotLifecycleRestIT.java]: Sure, I went ahead and changed it.

**Review by probakowski — APPROVED** [review]
LGTM

**Pull request merged**